src/Microsoft.SqlTools.ServiceLayer/DisasterRecovery/DisasterRecoveryService.cs

```diff
@@ -307,44 +315,51 @@ public void InitializeService(IProtocolEndpoint serviceHost)
+                bool supported = IsBackupRestoreOperationSupported(backupParams.OwnerUri, out connInfo);
+
+                if (supported && connInfo != null)
                 {
-                    BackupOperation backupOperation = CreateBackupOperation(helper.DataContainer, sqlConn, backupParams.BackupInfo);
+                    DatabaseTaskHelper helper = AdminService.CreateDatabaseTaskHelper(connInfo, databaseExists: true);
+                    // Open a new connection to use for the backup, which will be closed when the backup task is completed
+                    // (or an error occurs)
+                    SqlConnection sqlConn = ConnectionService.OpenSqlConnection(connInfo, "Backup");
+                    try
+                    {
+                        BackupOperation backupOperation = CreateBackupOperation(helper.DataContainer, sqlConn, backupParams.BackupInfo);
 
-                    // create task metadata
-                    TaskMetadata metadata = TaskMetadata.Create(backupParams, SR.BackupTaskName, backupOperation, ConnectionServiceInstance);
+                        // create task metadata
+                        TaskMetadata metadata = TaskMetadata.Create(backupParams, SR.BackupTaskName, backupOperation, ConnectionServiceInstance);
 
-                    SqlTask sqlTask = SqlTaskManagerInstance.CreateAndRun<SqlTask>(metadata);
-                    sqlTask.StatusChanged += (object sender, TaskEventArgs<SqlTaskStatus> e) =>
-                    {
-                        SqlTask sqlTask = e.SqlTask;
-                        if (sqlTask != null && sqlTask.IsCompleted)
+                        SqlTask sqlTask = SqlTaskManagerInstance.CreateAndRun<SqlTask>(metadata);
+                        sqlTask.StatusChanged += (object sender, TaskEventArgs<SqlTaskStatus> e) =>
                         {
-                            sqlConn.Dispose();
-                        }
-                    };
-                }
-                catch
-                {
-                    // Ensure that the connection is closed if any error occurs while starting up the task
-                    sqlConn.Dispose();
-                    throw;
+                            SqlTask sqlTask = e.SqlTask;
+                            if (sqlTask != null && sqlTask.IsCompleted)
+                            {
+                                sqlConn.Dispose();
+                            }
+                        };
+                        response.Result = true;
+                    }
+                    catch
+                    {
+                        // Ensure that the connection is closed if any error occurs while starting up the task
+                        sqlConn.Dispose();
+                        throw;
+                    }
                 }
-
             }
-            else
+            catch (Exception ex)
             {
-                response.Result = false;
+                Logger.Error(ex);
```

Comment:
Maybe explain here that the result will be sent back as false if an error occurs, so we just need to log for additional information.